pkg/images/images.go

```diff
@@ -144,8 +150,14 @@ func (r *Resolver) GetReference(i string, opts ...ResolverOpt) (name.Reference,
 		// Use override if set
 		ref = o
 	} else {
+		// Build the full prefix (registry/repoPrefix) while resolving default images
+		var prefix string
+		if r.repoPrefix != "" {
+			prefix = fmt.Sprintf("%s/%s", r.registry.Name(), r.repoPrefix)
+		}
+
 		// No override; get compile-time default
-		d, err := getDefaultImage(i)
+		d, err := getDefaultImage(i, prefix)
```

Comment:
nit: rather than building this here, and passing the prefix into `getDefaultImage`, I think this might fit better into `setRegistry` - since that's the function that's supposed to be responsible for applying the overrides from system-default-registry.